### DISCUSSION TITLE
Failed to build for iOS

### DISCUSSION THREAD

**Discussion by elhalabey** [codemagic-ci-cd/codemagic-docs#1979] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1- i getting started to deploy my first app on App Store and i didn't have a mac pc 
2 - i deploy ma app on google play store successfully 
3 - i read that i can deploy my app to app store without having mac with codemagic 
4 - after making all instruction the build failed with this error 
[!] Unable to find a target named `RunnerTests` in project `Runner.xcodeproj`, did find `Runner`.
i put a sample podfile 
but i faced another error :
Failed to build iOS app
Parse Issue (Xcode): Module 'image_picker_ios' not found
/Users/builder/clone/ios/Runner/GeneratedPluginRegistrant.m:11:8

Encountered error while archiving for device.

Build failed :|
Failed to build for iOS
[building_ios.log](https://github.com/codemagic-ci-cd/codemagic-docs/files/12384463/building_ios.log)

### Expected results

1 - Clear documentatio video or picture to get started with codemagic
2 - cannot deploy flutter app to playstore without testing on mac first
3 - sample files to put on my app to resolve all this issues

### Actual results

 i faced an error when building app says
[building_ios.log](https://github.com/codemagic-ci-cd/codemagic-docs/files/12384471/building_ios.log)
:
Failed to build iOS app
Parse Issue (Xcode): Module 'image_picker_ios' not found
/Users/builder/clone/ios/Runner/GeneratedPluginRegistrant.m:11:8

Encountered error while archiving for device.

### Build id (optional)

Failed to build iOS app Parse Issue (Xcode): Module 'image_picker_ios' not found /Users/builder/clone/ios/Runner/GeneratedPluginRegistrant.m:11:8  Encountered error while archiving for device.   Build failed :| Failed to build for iOS

**Comment by icarusdust** [comment, marked as answer]
Hi @elhalabey, please check these threads to see if any helps you: https://github.com/orgs/codemagic-ci-cd/discussions/1947 https://github.com/orgs/codemagic-ci-cd/discussions/1917#discussioncomment-6594275

For `Module 'image_picker_ios' not found`, check image_picker_ios version as it may require upgrade, check what Flutter version you are using, it may require upgrade or downgrade, check if the package is configured etc.
